src/algebra/big_operators/fin.lean

```diff
@@ -161,6 +161,26 @@ end partial_prod
 
 end fin
 
+namespace vector
+
+variables {n : ℕ} (a : α) (v : vector α n)
+
+lemma sum_filter_eq_nth_eq_count [decidable_eq α] :
+  ∑ i in univ.filter (λ i, a = v.nth i), 1 = v.to_list.count a :=
+begin
+  refine (finset.sum_filter _ _).trans (vector.induction_on v (sum_empty.trans ((congr_arg coe $
+    (list.count_nil a).symm).trans nat.cast_zero).symm) (λ n' x xs hxs, _)),
+  simp only [fin.sum_univ_succ, nth_cons_zero, to_list_cons, list.count_cons, ite_add,
+    add_comm 1, nth_cons_succ, zero_add, nat.succ_eq_add_one, @eq_comm _ x a],
+  congr; exact hxs,
+end
+
+lemma card_filter_univ_eq_nth_eq_count [decidable_eq α] :
+  (univ.filter $ λ i, a = v.nth i).card = v.to_list.count a :=
+((finset.sum_const 1).trans (by simp)).symm.trans (sum_filter_eq_nth_eq_count a v)
```

Comment:
I feel like it would be better to prove this result directly; after all, it makes no mention of big operators. I had a quick go at this, but found myself wanting the lemmas in #15385.

Reply:
Looking at that pull request, I was able to prove the more general
```
lemma card_filter_univ_fin (p : (fin n.succ) → Prop) [decidable_pred p] :
  (univ.filter p).card = (ite (p 0) 1 0) + (univ.filter (p ∘ fin.succ)).card :=
```
Using that, and then this lemma becomes just a special case of that with a quick induction. Once that PR is merged I can go ahead and make those changes here.

Reply:
@eric-wieser I've pushed these changes, the main lemma is now for a general predicate rather than vectors in particular.